security/src/main/java/io/micronaut/security/filters/SecurityFilter.java

```diff
@@ -76,23 +81,34 @@
      */
     private static final Integer ORDER = ServerFilterPhase.SECURITY.order();
 
-    protected final Collection<SecurityRule> securityRules;
+    /** Combined and sorted list of SecurityRule and ReactiveSecurityRule objects. */
+    protected final List<Ordered> orderedSecurityRules;
+
     protected final Collection<AuthenticationFetcher> authenticationFetchers;
 
     protected final SecurityConfiguration securityConfiguration;
 
     /**
-     * @param securityRules          The list of rules that will allow or reject the request
+     * @param securityRules The list of security rules that will allow or reject the request
+     * @param reactiveSecurityRules The list of reactive security rules that will allow or reject the request
      * @param authenticationFetchers List of {@link AuthenticationFetcher} beans in the context.
-     * @param securityConfiguration  The security configuration
+     * @param securityConfiguration The security configuration
      */
     @Inject
-    public SecurityFilter(Collection<SecurityRule> securityRules,
-                          Collection<AuthenticationFetcher> authenticationFetchers,
-                          SecurityConfiguration securityConfiguration) {
-        this.securityRules = securityRules;
+    public SecurityFilter(
+            Collection<SecurityRule> securityRules,
+            Collection<ReactiveSecurityRule> reactiveSecurityRules,
```

Comment:
This is a breaking change. This constructor can be deprecated and a new constructor created annotated with `@Inject`

Reply:
Updated.